mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+    if not os.path.exists(midi_path):
+        score, _ = split_score_annotations(fhandle)
+        score.write("midi", fp=midi_path)
+    return midi_path
+
+
+@io.coerce_to_string_io
+def load_roman_numerals(fhandle: TextIO, resolution=28):
+    """Load haydn op20 roman numerals data from a file
+
+    Args:
+        fhandle (str or file-like): path to roman numeral annotations
+
+    Returns:
+        List[dict]: musical roman numerals data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        figure = rn.figure
+        annotations.append({"time": time, "roman_numeral": figure})
+    return annotations
+
+
+def load_chords_base(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        List[dict]: musical chords data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    _, rna = split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        chord = rn.pitchedCommonName
+        annotations.append({"time": time, "chord": chord})
+    return annotations
+
+
+@io.coerce_to_string_io
+def load_chords(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        ChordData: chord annotations
+
+    """
+    chords = load_chords_base(fhandle, resolution)
+    start_times, end_times, chord_names = [0], [], [str(chords[0]["chord"])]
+    for ii, k in enumerate(chords):
+        if str(k["chord"]) != chord_names[-1]:
+            end_times.append(chords[ii]["time"] - 1)
+            start_times.append(chords[ii]["time"])
+            chord_names.append(str(chords[ii]["chord"]))
+    end_times.append(chords[-1]["time"])
+    return ChordData(np.array([start_times, end_times]).astype(float).T, chord_names)
+
+
+@io.coerce_to_string_io
+def load_chords_music21(fhandle: TextIO, resolution: int = 28):
+    """Load haydn op20 chords data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to chord annotations
+
+    Returns:
+        List[dict]: musical chords data and relative time (offset (Music21Object.offset) * resolution)
+
+    """
+    return load_chords_base(fhandle, resolution)
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The haydn op20 dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="haydn_op20",
+            track_class=Track,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            download_info=DOWNLOAD_INFO,
```

Comment:
you can delete this line, it defaults to empty :)

Reply:
done!